### PR TITLE
Pending directory search

### PR DESCRIPTION
**Pull request by robbierobs**
Adding pending directories to the search results

### CODE REVIEW

File: tendenci/apps/directories/views.py [hunk 1]
```diff
@@ -56,8 +56,41 @@ def details(request, slug=None, template_name="directories/view.html"):
...
+    filters = get_query_filters(user, 'directories.view_directory')
     directories = Directory.objects.filter(filters).distinct()
+
+    # Checking for view and change directory permission (staff)
+    can_view_directories = has_perm(user, 'directories.view_directory')
+    can_change_directories = has_perm(user, 'directories.change_directory')
+
+    can_view_pend = False
+    if can_view_directories and can_change_directories:
+        can_view_pend = True
```
Comment (yehudacohen) [inline tendenci/apps/directories/views.py:70]: `can_view_pend = can_view_directories and can_change_directories`

File: tendenci/apps/directories/views.py [hunk 2]
```diff
@@ -56,8 +56,41 @@ def details(request, slug=None, template_name="directories/view.html"):
...
+    # else, we're looking to see if the user is the creator or has
+    # membership with the directory
+    if not can_view_pend:
+        for dir in pending_dirs:
+            creator = False
+            member = False
+            dir_creator = dir.creator_username
+            if dir_creator == username:
+                creator = True
+
```
Comment (yehudacohen) [inline tendenci/apps/directories/views.py:85]: `creator = dir_creator == username`

File: tendenci/apps/directories/views.py [hunk 3]
```diff
@@ -56,8 +56,41 @@ def details(request, slug=None, template_name="directories/view.html"):
...
+    # Finding pending directories from the directories query we ran above
+    pending_dirs = directories.filter(status_detail='pending').distinct()
+
+    # Bypassing this check if the user already has permissions
+    # else, we're looking to see if the user is the creator or has
+    # membership with the directory
+    if not can_view_pend:
+        for dir in pending_dirs:
+            creator = False
+            member = False
```
Comment (yehudacohen) [inline tendenci/apps/directories/views.py:81]: we can just leave out this single line

File: tendenci/apps/directories/views.py [hunk 4]
```diff
@@ -56,8 +56,41 @@ def details(request, slug=None, template_name="directories/view.html"):
...
+    user = request.user
+    username = request.user.username
+    filters = get_query_filters(user, 'directories.view_directory')
     directories = Directory.objects.filter(filters).distinct()
+
+    # Checking for view and change directory permission (staff)
+    can_view_directories = has_perm(user, 'directories.view_directory')
+    can_change_directories = has_perm(user, 'directories.change_directory')
+
+    can_view_pend = False
```
Comment (yehudacohen) [inline tendenci/apps/directories/views.py:68]: This looks really nice. I'd like to be able to use this for other views as well. Can we expose the permission check as a Directory.is_editable_by(user) method within the directory model?

File: tendenci/apps/directories/views.py (hunk 1 above)
Comment (robbierobs) [reply tendenci/apps/directories/views.py]: Changed in the newest commit.

**Review by lgm527 — APPROVED** [review]
LGTM!

File: tendenci/apps/directories/views.py (hunk 2 above)
Comment (robbierobs) [reply tendenci/apps/directories/views.py]: changed in the newest commit

File: tendenci/apps/directories/views.py (hunk 3 above)
Comment (robbierobs) [reply tendenci/apps/directories/views.py]: Removed in the newest commit.

File: tendenci/apps/directories/views.py (hunk 4 above)
Comment (robbierobs) [reply tendenci/apps/directories/views.py]: Newest commit shows changes to:
`app/directories/utils.py` 

- can_view_pending_directories function to allow it to be callable in other models
- Imported the has_perm function to allow checking permissions against a user

and 

`app/directories/views.py`

- Condensed down code for a more pythonic flow

-

**Review by yehudacohen — APPROVED** [review]
Looks good to me!

**Pull request merged**